elastic/eui
# [EuiSuperDatePicker] Allow setting absolute dates without Enter key

## Summary

This PR adds the following two implicit date parsing triggers to EuiSuperDatePicker's absolute tabs, which allow users to set absolute dates without hitting the Enter key.

- On focus lost from the text input - i.e., blur event on it.
- Right before the absolute tab component's disposal - i.e., at `componentWillUnmount()`.

Steps to set 01:05:00.000 to 01:05:15.000:
1. Click start date.
2. Click `Absolute`.
3. Click `01:00`.
4. Select `0` to replace by mouse.
5. Type in `5`.
6. **No** need to hit Enter.
7. Select input text.
8. Copy it by Ctrl+C.
9. Click end date.
10. Click `Absolute`.
11. Select input text.
12. Paste it by Ctrl-V.
13. Select `00` to replace.
14. Type in `15`.
15. **No** need to hit Enter.
16. Click `Update`
![EuiSuperDatePicker](https://github.com/elastic/eui/assets/721858/ae6edb12-ba04-4217-ba3b-9ea40540aa1a)

Closes #7731

## QA

1. Go to https://eui.elastic.co/pr_7732/#/templates/super-date-picker
2. Open `Absolute` tab.
3. Click `Update`.
4. Open `Absolute` tab again.
5. Change the date text.
6. Click `Refresh` without hitting the Enter key.

- The date should be updated if the input date is compliant with the date format.
- The date shouldn't be updated if the input date is not compliant with the date format.
- When leaving the `Absolute` tab, even if the input date is not compliant with the date format, users shouldn't be notified about it.

### General checklist

- Browser QA
- Docs site QA
- Code quality checklist
- Release checklist
- Designer checklist
